src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -114,8 +137,10 @@ public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String d
                 this.getClass().getName(),
                 "fileService.lookupSnapshotFSItem");
         } catch (InterruptedException ex) {
+            currentDRSRequests--;
             throw new FileSystemExecutionException("Unexpected interruption during file system processing", ex);
         }
+        currentDRSRequests--; // TODO is there a better place for this?
```

Comment:
I'm not sure but I do want to point out that this isn't threadsafe.  Like if several threads are hitting this, there's no guarantee that the value of `currentDRSRequests` is the same than at the start of the method.  This could end up getting pretty gross.  Might be worth seeing if spring provides an endpoint annotation to do this...I'd be pretty sure shocked if they don't

Reply:
So I'm looking for an annotation to put on the variable that would make it consistent across threads?